Porting to Qgis 3

jorgealmerio opened the pull request:
Hi, Olivier

Sorry for the delay.

Here are the files changed for porting the plugin to the new python 3.xx format (Qgis 3)

I realized that "vectorbendertransformers.py" from here (master) is outdated comparing to the Qgis repository version, that's why there were so many code differences between my updated version and the master.

Thank you!

jorgealmerio commented:
Hi Olivier,

I did the all the uploads directly from github website. As I told you I am not github expert, I do not know exactly how to solve this line ending issues, but I saw a way to you see the real changes to the "vectorbenderdialog.py" by clicking on my second commint (I do not know why there is 3), please see pictures:
![vectorbenderdialog](https://user-images.githubusercontent.com/16169268/51632955-384cec00-1f2f-11e9-8bbd-4ccd354f94aa.png)

![image](https://user-images.githubusercontent.com/16169268/51633000-65999a00-1f2f-11e9-999f-0808f3c43c37.png)

olivierdalang commented on vectorbender.py:
```diff
@@ -103,28 +105,25 @@ def determineTransformationType(self):
             0 if no pairs Found
             1 if one pair found => translation
             2 if two pairs found => linear
-            3 if two pairs found => linear
```
I think this was mislabeled in the original code (should have been `3 if three pairs found => affine, 4 if four or more pairs found => bending`), but must be kept for affine transformation to still work.

jorgealmerio replied:
Done!

olivierdalang commented on vectorbender.py:
```diff
@@ -103,28 +105,25 @@ def determineTransformationType(self):
             return 1
         elif featuresCount == 2:
             return 2
-        elif featuresCount == 3:
```
same as above, we must keep the functionality for affine transform

jorgealmerio replied:
Done!

olivierdalang commented on vectorbender.py:
```diff
@@ -139,13 +138,10 @@ def run(self):
             self.dlg.displayMsg( "Loading delaunay mesh (%i points) ..." % len(self.ptsA) )
             QCoreApplication.processEvents()
             self.transformer = BendTransformer( pairsLayer, restrictToSelection, self.dlg.bufferValue() )
-        elif transType==3:
```
same as above...

jorgealmerio replied:
Done!

olivierdalang commented on vectorbendertransformers.py:
```diff
@@ -111,25 +111,19 @@ def __init__(self, pairsLayer, restrictToSelection):
 
 
         """
-
```
Revert these unrelated/unnecessary changes (for the whole file).

If you wish to improve the style, that's welcome, but should be part of another PR (so that this one really stays about porting to QGIS3)

jorgealmerio replied:
I tried to revert, I don't know if it's good now, because of number of spaces characters.

olivierdalang commented on vectorbendertransformers.py:
```diff
@@ -175,8 +166,8 @@ def __init__(self, pairsLayer, restrictToSelection):
 
-        return QgsPoint( self.a*p.x()+self.b*p.y()+self.c, self.d*p.x()+self.e*p.y()+self.f )
-
+        return QgsPointXY( self.a*p.x()+self.b*p.y()+self.c, self.d*p.x()+self.e*p.y()+self.f ) #Almerio: era QgsPoint
```
Remove these unnecessary `#Almerio: era QgsPoint` comments  (for the whole file)

jorgealmerio replied:
Done!

olivierdalang commented on metadata.txt:
```diff
@@ -23,8 +24,9 @@ email=olivier.dalang@gmail.com
 
 # Uncomment the following line and add your changelog entries:
 changelog=- 2014-05-22 - Version 0.0 : intial release
-- 2014-05-26 - Version 0.1 : linear and uniform transformation method, scipy dependency no more needed, better management of pinned points
```
Is this reformatting necessary for Qgis 3 compatibility ? If not, please revert.

If you wish to improve the style, that's welcome, but should be part of another PR (so that this one really stays about porting to QGIS3)

jorgealmerio commented:
1. Exception when running a transform (any type)
Line 269:
newGeom = QgsGeometry.fromPolyline( [geom[-1],geom[-1]] )
replaced by:
newGeom = QgsGeometry.fromPolylineXY( [geom[-1],geom[-1]] )

2. Dialog doesn't properly refresh
On vectorbenderdialog.py
Line 267:
if event.type() == QEvent.FocusIn:
replaced by:
if event.type() == QEvent.WindowActivate:

3. Deprecation warning (on plugin loading)
Line 152 from "objcreator.py" from Qgis is using a deprecated option it should be changed to:
plugin = open(filename, 'r')

It happens with some plugins (I am not so sure, by it depends on the way it calls the UI files), so it's not a Vectorbender warning.

olivierdalang commented:
Thanks @jorgealmerio , quickly tested today and it looks fine ! I'll push the new version to the repo ASAP
